smart_card_connector_app/src/window-apps-displaying.js

```diff
@@ -81,7 +96,24 @@ function updateAppView(knownAppsPromise, appIds, knownApps) {
       var text = knownApps && knownApps[i] ?
                  knownApps[i].name :
                  '<' + appIds[i] + '>';
-      var newElement = goog.dom.createDom('li', undefined, text);
+
+      var removePermission = goog.dom.createDom(
+          'button',
+          {'class': 'remove-permission-button',
+           'data-i18n-aria-label': 'removePermission',
+           'title': GSC.I18n.translateText('removePermission')},
+          '×');
+
+      // TODO(isandrk): What happens to the listeners of the 'dead' nodes?
+      goog.events.listen(
+          removePermission,
+          goog.events.EventType.CLICK,
+          removeApp.bind(undefined, appIds[i]));
+          // TODO(isandrk): ^ Is there some problem with this (renaming)? Possibly
+          // do the binding inside getPermissionsChecker (and rename it obviously).
+
+      var newElement = goog.dom.createDom(
+          'li', undefined, text, removePermission);
       goog.dom.append(appListElement, newElement);
     }
```

Comment:
It'd be also useful to render the denied items in the end of the list too (e.g. as grayed out or in some other way).
With that, the user would have an opportunity to undo her erroneous "Deny" selections.

(Note that we currently "workaround" this by not storing the denied decisions of the user in the storage - i.e. it's currently enough to restart the Chrome OS user session in order to flush all denied decisions. But after your change lands, we could also start preserving the denied decisions too, for better UX.)